docs/swarm/admin_guide.md

```diff
@@ -0,0 +1,238 @@
+services using **resource constraints** for *cpu* and *memory*.
+
+However, because manager nodes use the Raft consensus algorithm to replicate data
+in a consistent way, they are sensitive to resouce starvation. You should
+isolate managers in your swarm from processes that might block cluster
+operations like cluster heartbeat or leader elections.
+
+To avoid interference with manager node operation, you can drain manager nodes
+to make them unavailable as worker nodes:
+
+```bash
+docker node update --availability drain <NODE-ID>
+```
+
+When you drain a node, the scheduler reassigns any tasks running on the node to
+other available worker nodes in the cluster. It also prevents the scheduler from
+assigning tasks on the node.
+
+## Back up the cluster state
+
+Docker manager nodes store the cluster state and manager logs in the following
+directory:
+
+`/var/lib/docker/swarm/raft`
+
+Back up the raft data directory often so that you can use it in case of disaster
+recovery.
+
+You should never restart a manager node with the data directory from another
+node. For example, by copying the `raft` directory from one node to another. The
+data directory is unique to a node ID and a node can only use a given node ID
+once to join the swarm. (ie. Node ID space should be globally unique)
+
+To cleanly re-join a manager node to a cluster:
+
+1. Run `docker node demote <id-node>` to demote the node to a worker.
+2. Run `docker node rm <id-node>` before adding a node back with a fresh state.
+3. Re-join the node to the cluster using `docker swarm join`.
+
+In case of [disaster recovery](#disaster-recovery), you can take the raft data
+directory of one of the manager nodes to restore to a new swarm cluster.
+
+## Monitor swarm health
+
+You can monitor the health of Manager nodes by querying the docker `nodes` API
+in JSON format through the `/nodes` HTTP endpoint. Refer to the [nodes API documentation](https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api_v1.24.md#36-nodes)
+for more information.
+
+From the command line, run `docker node inspect <id-node>` to query the nodes. For instance, to query the reachability of the node as a Manager to other raft cluster members:
+
+```bash
+docker node inspect manager1 --format "{{json .ManagerStatus.Reachability}}"
+"reachable"
+```
+<!--TODO what are the other states and what do they mean?-->
+
+To query the status of the node as a worker that accept tasks:
+
+```bash
+docker node inspect manager1 --format "{{json .Status.State}}"
+"ready"
+```
+<!--TODO what are the other states and what do they mean?-->
+
+From those commands, we can see that `node01` is both at the status
+`reachable` as a manager and `ready` as a worker.
+
+An `unreachable` manager status on a Swarm manager means that this particular
+Manager is unreachable from other Managers. This means that action should be taken
+to restore this node to an healthy state (restarting the daemon, rebooting the machine...).
```

Comment:
usage: "swarm" lowercase, "manager" lowercase. Suggest:

An manager with `unreachable` status means the manager node is unreachable from other manager nodes. In this case you need to take action to restore an odd number of managers:
* Restart the daemon and see if the manager comes back as reachable.
* Reboot the machine.
* If neither restarting or rebooting work, you should add another manager node or promote a worker to be a manager node.